### PR TITLE
Proper check for value of min and max

### PR DESCRIPTION
**Pull request by mehdivk**
This PR addresses this bug https://github.com/TabDigital/strummer/issues/20

### CODE REVIEW

File: lib/matchers/number.js [hunk 1]
```diff
@@ -21,13 +23,13 @@ module.exports = function (opts) {
     }
 
     if (opts) {
-      if (opts.min && opts.max && (value < opts.min || value > opts.max)) {
```
Comment (nguyenchr) [inline lib/matchers/number.js]: Small optimisation but should we do

```js
    if (opts) {
      var hasMinValue = hasValue(opts.min);
      var hasMaxValue = hasValue(opts.max);
      if ( hasMinValue && hasMaxValue && (value < opts.min || value > opts.max)) {
        return 'should be a number between ' + opts.min + ' and ' + opts.max;
      }
      if (hasMinValue && value < opts.min) {
        return 'should be a number >= ' + opts.min;
      }
      if (hasMaxValue && value > opts.max) {
        return 'should be a number <= ' + opts.max;
      }
    }
```

**Comment by rprieto** [comment]
Nice catch! :+1: I think we might have similar bugs in `array` and `string`, should we have a look as part of this PR?

On top of this, maybe we could add some checks on `opts` before returning the matcher, for example asserting that `opts.min == null || typeof(opts.min) === 'number'`. This would not have any runtime performance if it's executed outside of the matcher function we return. See the [duration](https://github.com/TabDigital/strummer/blob/master/lib/matchers/duration.js) matcher, that asserts that the `min` and `max` durations are actually valid.

File: lib/matchers/number.js (hunk 1 above)
Comment (mehdivk) [inline lib/matchers/number.js]: @nguyenchr Nice optimisation.

**Comment by mehdivk** [comment]
@nguyenchr Your suggestion is applied. 
@rprieto Agree, I applied same pattern of validation to the number matcher.

**Comment by nguyenchr** [comment]
nice one :+1: 

is it worth making the same change to `array` and `string` matchers in this PR?

**Comment by mehdivk** [comment]
@rprieto @nguyenchr Good idea. I will extend this PR to fix other matchers as well.

**Comment by mehdivk** [comment]
@rprieto @nguyenchr Actually I don't think using this new method in `array` or `string` is a good thing. For it doesn't make sense if somebody says minimum length of array should be `0` or maximum length of array should be `0`. It's same thing for `string`. I feel we should leave them as it's now. Putting new logic doesn't change anything in mentioned matchers, it only brings more process to the matcher.

**Comment by rprieto** [comment]
Good point!

**Comment by nguyenchr** [comment]
:+1:

**Comment by mehdivk** [comment]
Per chat with @nguyenchr `changelog.md` is added to library to keep track of changelog for each version.

**Pull request merged**